components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -67,6 +69,7 @@ type odometry struct {
 	linearVelocity  r3.Vector
 	position        r3.Vector
 	orientation     spatialmath.EulerAngles
+	coordUpToDate   atomic.Bool
```

Comment:
Discussed with @randhid and came up with this method to avoid the [previously discussed](https://github.com/viamrobotics/rdk/pull/4121#discussion_r1651284671) race condition. Preserving a return of nil/NaN in the case where we do not have position data, Reconfigure and DoCommand will now wait for trackPosition to update coord (or for a ctx to timeout) so that stale data is never returned.

Rand asked that I tag @martha-johnston to look this over.